### PR TITLE
Add types to databse layer abstraction

### PR DESCRIPTION
**Pull request by siwane**
As preparation, for google spanner implementation, please find optional `$types` args added to persistence methods

### CODE REVIEW

File: common/persistence/sql/dbal/class.Driver.php [hunk 1]
```diff
@@ -150,38 +155,40 @@ public function getSchemaManager()
     {
         return new common_persistence_sql_dbal_SchemaManager($this->connection->getSchemaManager());
     }
-    
-   
+
```
Comment (julien-sebire) [inline common/persistence/sql/dbal/class.Driver.php]: Double blank line.

File: common/persistence/class.SqlPersistence.php [hunk 2]
```diff
@@ -14,70 +15,79 @@
...
+
     /**
+     * Execute a statement.
+     *
+     * @param string $statement
+     * @param array $params
+     * @param array $types
+     * @return int number of updated rows
+     */
+    public function exec($statement, array $params = [], array $types = [])
```
Comment (mike-ionut-mihai-sandu-tao) [inline common/persistence/class.SqlPersistence.php]: do we need to alter and update any of the tests?

File: manifest.php [hunk 3]
```diff
@@ -34,7 +34,7 @@
     'label' => 'Generis Core',
     'description' => 'Core extension, provide the low level framework and an API to manage ontologies',
     'license' => 'GPL-2.0',
-    'version' => '12.8.0',
+    'version' => '12.9.0',
```
Comment (julien-sebire) [inline manifest.php]: What about the version bump?

File: common/persistence/class.SqlPersistence.php (hunk 2 above)
Comment (siwane) [reply common/persistence/class.SqlPersistence.php]: after checking nope, that may requires integration test

**Pull request merged**